packages/components/pie-text-input/src/index.ts

```diff
@@ -201,7 +203,12 @@ export class PieTextInput extends FormControlMixin(RtlMixin(LitElement)) impleme
                 data-pie-status=${ifDefined(status)}
                 ?data-pie-disabled=${live(disabled)}
                 ?data-pie-readonly=${readonly}>
-                <slot name="leading"></slot>
+                <!-- The reason for separate slots for icons and text is that we cannot programmatically be aware of the
+                HTML used inside of the slot without breaking SSR in consuming applications (icons need to use different colours than text content)
+                Therefore, we provide two slots and advise consumers do not attempt to use both leading/trailing at the same time
+                as this would violate the design guidelines for this component. -->
+                <slot name="leadingIcon"></slot>
```

Comment:
tldr; the icon in a slot needs to be a different colour from text in the same slot. I could not find a way to do this in CSS with them sharing the same slot name. There are ways in lit to be aware of what HTML is in the slot and apply different classes, but as the comment says this seems to break SSR in apps such as NextJS